src/main/java/com/develhope/spring/vehicle/service/VehicleService.java

```diff
@@ -0,0 +1,19 @@
+package com.develhope.spring.vehicle.service;
+
+import com.develhope.spring.vehicle.repositories.VehicleRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class VehicleService {
+
+    @Autowired
+    private VehicleRepository vehicleRepository;
+
+
+
+
+    public void deleteVehicle(long id) {

```

Comment:
Devi assicurarti che solo l'admin possa fare questa operazione.  Quindi devi o passargli uno user da cui prendi il ruolo e verifichi che sia un admin. O un id utente e ugualmente tramite id trovi lo user e verifichi che abbia il ruolo di admin prima di procedere con l'operazione. Inoltre devi controllare il caso in cui tramite id non trovi nulla. Sia se non trovi un utente, sia se non trovi un veicolo. Altrimenti il programma da un 500 internal server error. Quindi devi lanciare le eccezioni qua, e nel controller tramite exception handler come ha fatto vedere Lorenzo nel pomeriggio, le vai a gestire.